### PR TITLE
Fix some clang-tidy lints for Linux host_debug

### PR DESCRIPTION
**Pull request by zanderso**
This PR adds some curly braces, some instances of `explicit`, and converts some C style casts to C++ style casts.

For https://github.com/flutter/flutter/issues/93576

### CODE REVIEW

File: shell/common/shell_unittests.cc [hunk 1]
```diff
@@ -1877,7 +1877,7 @@ TEST_F(ShellTest, CanConvertToAndFromMappings) {
   const size_t buffer_size = 2 << 20;
 
   uint8_t* buffer = static_cast<uint8_t*>(::malloc(buffer_size));
-  ASSERT_NE(buffer, nullptr);
+  ASSERT_NE(buffer, nullptr);  // NOLINT(clang-analyzer-unix.Malloc)
```
Comment (dnfield) [inline shell/common/shell_unittests.cc:1880]: I think assert_true would avoid the ignore

**Review by dnfield — APPROVED** [review]
Lgtm w nit

**Comment by zanderso** [comment]
@CaseyHillers Why is this PR still blocked?

**Comment by jmagman** [comment]
This one also mysteriously flipped back to `master`.

**Pull request merged**